tests/reg_tests/test_DVGeometryMulti.py

```diff
@@ -315,6 +315,139 @@ def twist(val, geo, nRefAxPts=nRefAxPts):
+            "part_22_1d",
+            "part_23_1d",
+        ]
+
+        # Define a name for the point set
+        ptSetName = "test_set"
+
+        # Define a test point set
+        pts = np.array(
+            [
+                [1.0, -0.4, 0.5],  # curve 22
+                [1.0, -0.2, 0.5],  # curve 22
+                [1.0, 0.2, 0.5],  # curve 23
+                [1.0, 0.4, 0.5],  # curve 23
+            ]
+        )
+
+        # Compute the processor sizes with integer division
+        sizes = np.zeros(comm.size, dtype="intc")
+        nPtsGlobal = pts.shape[0]
+        sizes[:] = nPtsGlobal // comm.size
+
+        # Add the leftovers
+        sizes[: nPtsGlobal % comm.size] += 1
+
+        # Compute the processor displacements
+        disp = np.zeros(comm.size + 1, dtype="intc")
+        disp[1:] = np.cumsum(sizes)
+
+        # Split up the point set
+        localPts = pts[disp[comm.rank] : disp[comm.rank + 1]]
+
+        # Add the intersection between box1 and box2
+        DVGeo.addIntersection(
+            "box1",
+            "box2",
+            dStarA=1.0,
+            dStarB=0.15,
+            featureCurves=featureCurves,
+            project=True,
+            includeCurves=True,
+            slidingCurves=slidingCurves,
+            curveEpsDict=curveEpsDict,
+        )
+
+        # Add a few design variables
+        DVGeoDict = DVGeo.getDVGeoDict()
+        for comp in comps:
+            # Create reference axis
+            nRefAxPts = DVGeoDict[comp].addRefAxis("box", xFraction=0.5, alignIndex="j", rotType=4)
+            nTwist = nRefAxPts - 1
+
+            # Set up a twist variable
+            def twist(val, geo, nRefAxPts=nRefAxPts):
+                for i in range(1, nRefAxPts):
+                    geo.rot_z["box"].coef[i] = val[i - 1]
+
+            DVGeoDict[comp].addGlobalDV(dvName=f"{comp}_twist", value=[0] * nTwist, func=twist)
+
+        # Add the point set
+        DVGeo.addPointSet(localPts, ptSetName, comm=comm, applyIC=True)
+
+        # Apply twist to box 2
+        dvDict = DVGeo.getValues()
+        dvDict["box2_twist"] = 10
+        DVGeo.setDesignVars(dvDict)
+
+        # Update the point set
+        ptsUpdated = DVGeo.update(ptSetName)
+
+        # Create the send buffer
+        procPoints = ptsUpdated.flatten()
+        sendbuf = [procPoints, sizes[comm.rank] * 3]
+
+        # Create the receiving buffer
+        globalPoints = np.zeros(nPtsGlobal * 3)
+        recvbuf = [globalPoints, sizes * 3, disp[0:-1] * 3, MPI.DOUBLE]
+
+        # Allgather the updated coordinates
+        comm.Allgatherv(sendbuf, recvbuf)
+
+        # Reshape into a nPtsGlobal, 3 array
+        ptsUpdated = globalPoints.reshape((nPtsGlobal, 3))
+
+        # Test that the X and Z coordinates are unchanged and Y coordinates are changed
+        np.testing.assert_allclose(pts[:, 0], ptsUpdated[:, 0], rtol=1e-4, atol=1e-10)
+        np.testing.assert_allclose(pts[:, 2], ptsUpdated[:, 2], rtol=1e-4, atol=1e-10)
```

Comment:
The `rtol` for these should be tighter than 1e-4, right? I would expect the coordinates to match to 8 digits at least

Reply:
I'll make it tighter. I matched the other tests for consistency but I'll change it, you are right that it should match essentially to numerical precision.